src/hasSubObject.ts

```diff
@@ -25,10 +75,10 @@ type HasSubObjectGuard<T, S> = Simplify<
  * @dataFirst
  * @category Guard
  */
-export function hasSubObject<T, S extends Partial<T>>(
-  data: T,
-  subObject: S,
-): data is HasSubObjectGuard<T, S>;
+export function hasSubObject<
+  T extends Record<string, unknown>,
```

Comment:
**question:** why `string` as key? shouldn't this be any `PropertyKey`

Reply:
@cjquines Good catch, will fix it.